### PR TITLE
Add ceiling, floor, `INT`, and `MOD` formula functions

### PR DESCRIPTION
**Pull request by HactarCE**
Depends on `next-connections` branch

This includes:
- `CEILING` and `FLOOR`
- `CEILING.MATH` and `FLOOR.MATH`
- `INT` and `MOD`

### CODE REVIEW

File: quadratic-core/src/formulas/functions/mathematics.rs [hunk 1]
```diff
@@ -245,12 +296,70 @@ mod tests {
...
+            ("-3.5", "-2", Ok("-4")),
+        ];
+        for (n, increment, expected) in test_cases {
+            let formula = format!("CEIL({n}, {increment})");
+            match expected {
+                Ok(ok) => assert_eq!(ok, eval_to_string(&g, &formula)),
+                Err(err) => assert_eq!(err, eval_to_err(&g, &formula).msg),
+            }
+        }
+    }
```
Comment (davidkircos) [inline quadratic-core/src/formulas/functions/mathematics.rs:332]: Do you test leaving the increment argument undefined?

Comment (HactarCE) [reply quadratic-core/src/formulas/functions/mathematics.rs:332]: Both arguments are required, as per Excel.

File: quadratic-core/src/formulas/functions/mathematics.rs [hunk 2]
```diff
@@ -65,6 +65,153 @@ fn get_functions() -> Vec<FormulaFunction> {
                 number.sqrt()
             }
         ),
+        formula_fn!(
+            /// Rounds a number up to the next multiple of `increment`. If
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/mathematics.rs:69]: @luke-quadratic many docs to review in this file

**Review by ddimaria — APPROVED** [review]
Lots of great additions 🙌

File: quadratic-core/src/formulas/functions/mathematics.rs (hunk 1 above)
Comment (HactarCE) [reply quadratic-core/src/formulas/functions/mathematics.rs:426]: Actually you're right, I should still add a test for this to make sure it gives the right error message.

**Comment by HactarCE** [comment]
Oops, in the commit title I said `INT` and `SUM` but it should be `INT` and `MOD`

**Pull request merged**